pom.xml

```diff
@@ -139,6 +139,28 @@
   </dependencies>
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M2</version> <!-- FIXME should come from the parent POM-->
+        <executions>
+          <execution>
+            <id>enforce-assertj-exclusion</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <bannedDependencies>
+                  <excludes>
+                    <exclude>org.assertj:assertj-core</exclude>
```

Comment:
the policy in assertj-core is no dependency at all for the main code which I think  could be translated with (TBC)
```
*:*:*:jar:compile:*
```

Reply:
`bytebuddy` is currently a compile dependency but I guess is fine, so I would add it as allowed. I would still exclude `assertj-core` in case of test scope.

Reply:
Indeed, but that's the only one and it is shaded, I think we can just include it and exclude everything else.

Reply:
@scordio juts added you as a collaborator on assertj parent pom, just commit directly the maven-enforcer-plugin declaration in there

Reply:
Thanks!

Not sure I got you correctly, summarizing I'd put just the plugin version in the parent pom and the inclusions/exclusions should stay here in the core project, correct?

BTW I didn't understand why the build is failing and how it's related to my changes, locally that test seems fine both under maven and in the IDE.

Reply:
I was able to reproduce it locally, the issue comes from the `ConfigurationForTests` loaded when the first Assert instance is created (this create a `WritableAssertionInfo` which calls `CONFIGURATION_PROVIDER.representation()` which loads `ConfigurationForTests`).

`ConfigurationForTests` disable filtering AssertJ elements from the stack trace when the failing test itself wanted to enable it.

Why it fails just now, I have no clue as I have added it in f258fa9604540d9587a41faf8bd8fb492d5fa920 a few month ago.

Anyway this is actually a bug since the intention of registering a configuration was that it was applied before any tests was run, it turns out that it is loaded when the first assertion is built which is after the first test started.

I think I'll disable the Configuration auto discovery for the moment.

It's actually lucky as I was about to release 3.13.0 :sweat_smile:

Reply:
@joel-costigliola plugin config updated and conflict resolved, I think this PR should be ready now